### ISSUE TITLE
ISR - `pageProps` in the `_app` is empty

### ISSUE THREAD

**Issue by GabenGar** [vercel/next.js#36799]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

 Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10
  Binaries:
      Node: 12.22.8
      npm: N/A
      Yarn: N/A
      pnpm: N/A
  Relevant packages:
      next: 12.1.7-canary.3
      react: 18.1.0
      react-dom: 18.1.0

### What browser are you using? (if relevant)

firefox 100.0 (64-bit)

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

`pageProps` is absent on the `_app` for ISR pages.

### Expected Behavior

`pageProps` being there.

### To Reproduce

```sh
git clone https://github.com/GabenGar/repros.git repros
cd repros/nextjs/isr-build-page-props
npm install
npm build
```

Ignore errors related to `_error` pages, the point is `pages/page/[id]` doesn't have `pageProps` despite it being passed in `getStaticProps()` (uncomment the logging line to confirm it does in fact return the `props` with values).

**Comment by icyJoseph** [comment]
Hi,

Sorry for this opening, but this is expected behavior.

In `pages/page/[id]` change `getStaticPaths` to use `fallback: "blocking"`, and the error goes away.

Why would that be though?

When we use `fallback: true`, the server sends a loading page to the user agent, and that page is also generated during build time. If you want to have fallback `true`, do this change:

```tsx

function Page({
  locale,
  page,
}: InferGetStaticPropsType<typeof getStaticProps>) {
  const router = useRouter();

  if (router.isFallback) return <div>Loading...</div>;

  return (
    <div>
      <p>{locale}</p>
      <article>
        <header>{page.id}</header>
        <section>{page.description}</section>
      </article>
    </div>
  );
}
```

Since ISR are generated from next/link, for most cases, the fallback page won't be shown. Most of the item, only to those who come to a page that has not been generated yet.

**Comment by icyJoseph** [comment]
> I do have a similar logic in my code, but I handle it in a [separate page component](https://github.com/GabenGar/next-js-site/blob/256aa60ab3479aa41d71d9752200917a26aaa781/src/components/pages/base.tsx#L34). Another problem is this crash happens only at build time, which makes it pretty hard to debug.

Yeah, I could agree with that, but that's what I like about the Next.js build step, it is like an integration test, for free. However, when there's a lot of moving parts, perhaps the last thing to come to mind is that the fallback page also needs to be rendered.

It might not be so obvious, but it exists in the documentation, and it works as expected.

**Comment by GabenGar** [comment]
It absolutely isn't obvious because it changes the behaviour of the code entirely on build. 
Your code example won't solve the problem because the crash happens in the [`_app`](https://github.com/GabenGar/repros/blob/main/nextjs/isr-build-page-props/src/pages/_app.tsx).

**Comment by icyJoseph** [comment]
> It absolutely isn't obvious because it changes the behaviour of the code entirely on build. Your code example won't solve the problem because the crash happens in the [`_app`](https://github.com/GabenGar/repros/blob/main/nextjs/isr-build-page-props/src/pages/_app.tsx).

You are making it crash on `_app` because you are ignoring the empty fallback render. You cannot assume that `props.locale` will be present in all cases.

**Comment by GabenGar** [comment]
Isn't fallback render a client-only feature though? If I am going to put the fallback logic right into the `_app` then server-side rendered pages will be affected too.

**Comment by icyJoseph** [comment]
Why would you need to do this logic in `_app`? You can do it at page level.

The fallback is there for when a page has not been generated yet. For example if someone navigates directly to `/blog/posts/123` but it is the first visitor ever, then the fallback is shown (if fallback is true), while the page is generated.

**Comment by GabenGar** [comment]
I need to pass some props for SEO (which also have to be calculated server-side) and doing it at page level means destructuring props at every page and passing them to SEO module. Which is as far from DRY as it can get.
To be honest ISR should've been a separate function, it has way too many gotchas to be just an option of static generation.
